src/plugins/ui_core_views/cell_evaluation/dependencies_r_tree.ts

```diff
@@ -0,0 +1,126 @@
+import { deepEquals } from "../../../helpers/misc";
+import { BoundedRange, UID } from "../../../types";
+import { RTreeBoundingBox, RTreeItem, SpreadsheetRTree } from "./r_tree";
+import { RangeSet } from "./range_set";
+
+interface RangeSetItem {
+  boundingBox: RTreeBoundingBox;
+  data: RangeSet;
+}
+
+type RTreeRangeItem = RTreeItem<BoundedRange>;
+
+/**
+ * R-Tree of ranges, mapping zones (r-tree bounding boxes) to ranges (data of the r-tree item).
+ * Ranges associated to the exact same bounding box are grouped together
+ * to reduce the number of nodes in the R-tree.
+ */
+export class DependenciesRTree {
+  private readonly rTree: SpreadsheetRTree<RangeSet>;
+
+  constructor(items: RTreeRangeItem[] = []) {
+    const compactedBoxes = groupSameBoundingBoxes(items);
+    this.rTree = new SpreadsheetRTree(compactedBoxes);
+  }
+
+  insert(item: RTreeRangeItem) {
+    const data = this.rTree.search(item.boundingBox);
+    const exactBoundingBox = data.find((d) => deepEquals(d.boundingBox, item.boundingBox));
```

Comment:
maybe replace all deepEquals with a straight comparison since we know the structure of a boundingbox ?

Reply:
Seems worth it 
https://jsperf.app/pifawe